pygmt/src/nearneighbor.py

```diff
@@ -0,0 +1,132 @@
+"""
+nearneighbor - Grid table data using a "Nearest neighbor" algorithm
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    data_kind,
+    dummy_context,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    I="spacing",
+    R="region",
+    V="verbose",
+    G="outfile",
+    a="aspatial",
+    f="coltypes",
+    r="registration",
+    S="search_radius",
+    E="empty",
+    N="sectors",
```

Comment:
Sort by [ASCIIbetical](https://en.wikipedia.org/wiki/ASCII#Character_order) order. Also, I'm debating whether to change the `search_radius` parameter to `searchradius` per convention at https://www.pygmt.org/v0.4.0/contributing.html#code-style (*Do not use an underscore to separate words bridged only by consonants*), though the GMT docs at https://docs.generic-mapping-tools.org/6.2/nearneighbor.html#s and GMT.jl uses `search_radius` with an underscore.

```suggestion
    E="empty",
    G="outfile",
    I="spacing",
    N="sectors",
    R="region",
    S="search_radius",
    V="verbose",
    a="aspatial",
    f="coltypes",
    r="registration",
```

Reply:
Yes, I thought aligning with GMT was probably best for `search_radius`.